solace/src/main/java/org/apache/apex/malhar/solace/AbstractSolaceBaseInputOperator.java

```diff
@@ -0,0 +1,319 @@
+import com.solacesystems.jcsmp.JCSMPException;
+import com.solacesystems.jcsmp.JCSMPFactory;
+import com.solacesystems.jcsmp.JCSMPProperties;
+import com.solacesystems.jcsmp.JCSMPReconnectEventHandler;
+import com.solacesystems.jcsmp.JCSMPSession;
+import com.solacesystems.jcsmp.SessionEvent;
+import com.solacesystems.jcsmp.SessionEventArgs;
+import com.solacesystems.jcsmp.SessionEventHandler;
+import com.solacesystems.jcsmp.XMLMessageListener;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.InputOperator;
+import com.datatorrent.api.Operator;
+import com.datatorrent.api.Operator.CheckpointListener;
+import com.datatorrent.common.util.BaseOperator;
+import com.datatorrent.lib.io.IdempotentStorageManager;
+import com.datatorrent.netlet.util.DTThrowable;
+
+@SuppressWarnings("unused")
+public abstract class AbstractSolaceBaseInputOperator<T> extends BaseOperator implements
+    InputOperator, Operator.ActivationListener<Context.OperatorContext>, CheckpointListener
+{
+
+  private static final Logger logger = LoggerFactory.getLogger(AbstractSolaceBaseInputOperator.class);
+
+  @NotNull
+  protected JCSMPProperties properties = new JCSMPProperties();
+  protected String connectRetries;
+  protected String reconnectRetries;
+  protected String unackedMessageLimit;
+
+
+  //protected IdempotentStorageManager idempotentStorageManager = new IdempotentStorageManager.FSIdempotentStorageManager();
+  protected FSOpsIdempotentStorageManager idempotentStorageManager = new FSOpsIdempotentStorageManager();
+
+  protected transient JCSMPFactory factory;
+  protected transient JCSMPSession session;
+
+  protected transient Consumer consumer;
+  protected transient Consumer reliableConsumer;
+
+  protected transient int operatorId;
+  protected long windowId;
+  protected transient long lastCompletedWId;
+
+  protected transient int emitCount;
+
+  protected transient volatile boolean DRFailover = false;
+  protected transient volatile boolean TCPDisconnected = false;
+
+  protected transient BlockingQueue<BytesXMLMessage> unackedMessages; // hosts the Solace messages that need to be acked when the streaming window is OK to remove
+  protected LinkedList<Long> inFlightMessageId = new LinkedList<Long>(); //keeps track of all in flight IDs since they are not necessarily sequential
+  protected transient ArrayBlockingQueue<BytesXMLMessage> arrivedTopicMessagesToProcess;
+
+  //protected transient com.solace.dt.operator.DTSolaceOperatorInputOutput.ArrayBlockingQueue<BytesXMLMessage> arrivedMessagesToProcess;
+
+  private transient ReconnectCallbackHandler rcHandler = new ReconnectCallbackHandler();
+
+  private transient CallbackMessageHandler cbHandler = new CallbackMessageHandler();
+
+  int spinMillis;
+
+  protected transient int reconnectRetryMillis = 0;
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+
+    operatorId = context.getId();
+    logger.info("OperatorID from Base class: {}", operatorId);
```

Comment:
Why logger.info() and not logger.debug()? Also the "from Base class" comment is not clear. Do you mean AbstractSolaceBaseInputOperator class? Is this logger line useful?